### PR TITLE
update: PORTAL-1238 Convert Portal Backend API App Code

### PR DESCRIPTION
**Pull request by ian-sawyer**
This PR converts the previous TypeScript/Node.js api code to a Kotlin/Spring MVC Rest API. Please review the updated `README.md`. Make sure to follow the new documentation. You will need to set environment variables for the new api to pickup. Please reach out to me for any questions you have.

I tested and successfully retrieved an auth_token with the new api through the grant type authorize code oauth flow. This is the primary test for this PR change because the supplemental api is still in development. Testing the destination/status endpoint is not possible right now.

### CODE REVIEW

File: api/.gitignore [hunk 1]
```diff
@@ -1,48 +1,40 @@
...
-/packages
-/TestResults
-
-/tools/NuGet.exe
-/App_Data
-/secrets
-/data
-.secrets
-appsettings.json
-local.settings.json
```
Comment (tbolt) [inline api/.gitignore:23]: Do we want this file checked in to the codebase? I have it locally and it will be checked in next commit I do unless I `rm` it.

**Review by tbolt — COMMENTED** [review]
Looks good to me, granted I haven't worked with java/kotlin in years so I wouldn't trust my opinion alone for code review. 

I did review it though and could verify that traces of the old TS code is gone and the new documentation looks good.

I was able to remove `/node_modules` and `/dist` too.

I will let @sublettk do a final review

File: api/.gitignore (hunk 1 above)
Comment (ian-sawyer) [reply api/.gitignore:23]: No, we shouldn't check it in to the codebase.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Data Exchange Portal
 
-This is the web application for the data upload management dashboard for DEX. It is built using ReactJS and Vite. It is deployed to one of three environments in the [ocio-dmz-dex](https://portal.azure.com/#@cdc.onmicrosoft.com/resource/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/OCIO-API-PRD/providers/Microsoft.Web/staticSites/ocio-dmz-dex/staticsite) Azure Static Web App. It also uses managed Azure Functions for the middleware API endpoints.
+This is the web application for the data upload management dashboard for DEX. It is built using ReactJS and Vite. The backend `api` application utlizes Kotlin/Spring MVC.
```
Comment (sublettk) [inline README.md:3]: Low Priority

typo: utlizes --> utilizes

File: api/src/main/kotlin/com/dexportalapi/DeXPortalAPI/web/MainController.kt [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.reactive.function.BodyInserters
+import org.springframework.web.reactive.function.client.*
+import org.springframework.web.server.ResponseStatusException
+
+/**
+ * TODO:
+ * - add CustomInternalServerErrorException
+ * - add CustomBadRequestException
```
Comment (sublettk) [inline api/src/main/kotlin/com/dexportalapi/DeXPortalAPI/web/MainController.kt:21]: Did you still want to do this?

**Review by sublettk — APPROVED** [review]
Amazing work, Ian!

I just have some low priority things to address and then you're good to merge!

File: api/src/main/kotlin/com/dexportalapi/DeXPortalAPI/web/MainController.kt (hunk 3 above)
Comment (ian-sawyer) [reply api/src/main/kotlin/com/dexportalapi/DeXPortalAPI/web/MainController.kt:21]: I think it would be a good idea to in the future.

**Pull request merged**